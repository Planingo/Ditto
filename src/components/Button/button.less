.storybook-button {
  .ant-radio-group  {
    .active {
      background: var(--PRIMARY) !important;
      border-color: var(--PRIMARY) !important;
      color: var(--WHITE) !important;
    }

    .unactive{
      background: var(--WHITE) !important;
      border-color: var(--PRIMARY) !important;
      color: var(--PRIMARY) !important;
    }

    svg {
      font-size: 1em !important;
    }
  }

  button {
    display: flex;
    align-items: center;

    background: var(--PRIMARY) !important;
    border-color: var(--PRIMARY) !important;
    color: var(--WHITE-100) !important;
    
    * {
      color: var(--WHITE-100) !important;
    }

    &:hover {
      background: var(--PRIMARY-080) !important;
      border-color: var(--PRIMARY-080) !important;
    }

    &:active {
      background: var(--PRIMARY-140) !important;
      border-color: var(--PRIMARY-140) !important;
    }

    &[data-link] {
      background: transparent !important;
      border: none !important;
      color: var(--PRIMARY) !important;
      
      * {
        color: var(--PRIMARY) !important;
      }

      &:hover {
        * {
          color: var(--PRIMARY-080) !important;
        }
      }

      &:active {
        * {
          color: var(--PRIMARY-140) !important;
        }
      }

      &[disabled] {
        background: var(--WHITE-020) !important;  
        border-color: var(--PRIMARY-060) !important;
        
        * {
          color: var(--PRIMARY-060) !important;
        } 
      }
    }

    &[disabled] {
      background: var(--PRIMARY-020) !important;
      border-color: var(--PRIMARY-060) !important;
      
      * {
        color: var(--PRIMARY-060) !important;
      } 
    }
  
    &[data-ghost] {
      background: transparent !important;
      border-color: var(--PRIMARY) !important;
      color: var(--PRIMARY) !important;
      
      * {
        color: var(--PRIMARY) !important;
      }

      &:hover {
        border-color: var(--PRIMARY-080) !important;
        color: var(--PRIMARY-080) !important;

        * {
          color: var(--PRIMARY-080) !important;
        }
      }

      &:active {
        border-color: var(--PRIMARY-140) !important;
        color: var(--PRIMARY-140) !important;

        * {
          color: var(--PRIMARY-140) !important;
        }
      }

      &[disabled] {
        background: var(--WHITE-020) !important;  
        border-color: var(--PRIMARY-060) !important;
        
        * {
          color: var(--PRIMARY-060) !important;
        } 
      }
    }

    &[data-danger] {
      background: var(--CANCEL-ERROR) !important;
      border-color: var(--CANCEL-ERROR) !important;
      color: var(--WHITE-100) !important;
      
      * {
        color: var(--WHITE-100) !important;
      }

      &:hover {
        background: var(--RED-080) !important;
        border-color: var(--RED-080) !important;
      }


      &[data-link] {
        background: transparent !important;
        border: none !important;
        color: var(--RED) !important;
        
        * {
          color: var(--RED) !important;
        }

        &:hover {
          * {
            color: var(--RED-080) !important;
          }
        }

        &:active {
          * {
            color: var(--RED-140) !important;
          }
        }

        &[disabled] {
          background: var(--WHITE-020) !important;  
          border-color: var(--RED-060) !important;
          
          * {
            color: var(--RED-060) !important;
          } 
        }
      }

      &:active {
        background: var(--RED-140) !important;
        border-color: var(--RED-140) !important;
      }

      &[disabled] {
        background: var(--RED-020) !important;  
        border-color: var(--RED-060) !important;
        
        * {
          color: var(--RED-060) !important;
        } 
      }

      &[data-ghost] {
        background: transparent !important;
        border-color: var(--CANCEL-ERROR) !important;
        color: var(--CANCEL-ERROR) !important;
        
        * {
          color: var(--CANCEL-ERROR) !important;
        }

        &:hover {
          border-color: var(--RED-080) !important;
          color: var(--RED-080) !important;
  
          * {
            color: var(--RED-080) !important;
          }
        }
  
        &:active {
          border-color: var(--RED-140) !important;
          color: var(--RED-140) !important;
  
          * {
            color: var(--RED-140) !important;
          }
        }
  
        &[disabled] {
          background: var(--WHITE-020) !important;  
          border-color: var(--RED-060) !important;
          
          * {
            color: var(--RED-060) !important;
          } 
        }
      }
    }

    &.cta {
      background: var(--SECONDARY) !important;
      border-color: var(--SECONDARY) !important;
      color: var(--WHITE-100) !important;
      
      * {
        color: var(--WHITE-100) !important;
      }

      &:hover {
        background: var(--SECONDARY-080) !important;
        border-color: var(--SECONDARY-080) !important;
      }


      &[data-link] {
        background: transparent !important;
        border: none !important;
        color: var(--SECONDARY) !important;
        
        * {
          color: var(--SECONDARY) !important;
        }

        &:hover {
          * {
            color: var(--SECONDARY-080) !important;
          }
        }

        &:active {
          * {
            color: var(--SECONDARY-140) !important;
          }
        }

        &[disabled] {
          background: var(--WHITE-020) !important;  
          border-color: var(--SECONDARY-060) !important;
          
          * {
            color: var(--SECONDARY-060) !important;
          } 
        }
      }

      &:active {
        background: var(--SECONDARY-140) !important;
        border-color: var(--SECONDARY-140) !important;
      }

      &[disabled] {
        background: var(--SECONDARY-020) !important;  
        border-color: var(--SECONDARY-060) !important;
        
        * {
          color: var(--SECONDARY-060) !important;
        } 
      }

      &[data-ghost] {
        background: transparent !important;
        border-color: var(--SECONDARY) !important;
        color: var(--SECONDARY) !important;
        
        * {
          color: var(--SECONDARY) !important;
        }

        &:hover {
          border-color: var(--SECONDARY-080) !important;
          color: var(--SECONDARY-080) !important;

          * {
            color: var(--SECONDARY-080) !important;
          }
        }

        &:active {
          border-color: var(--SECONDARY-140) !important;
          color: var(--SECONDARY-140) !important;

          * {
            color: var(--SECONDARY-140) !important;
          }
        }

        &[disabled] {
          background: var(--WHITE-020) !important;  
          border-color: var(--SECONDARY-060) !important;
          
          * {
            color: var(--SECONDARY-060) !important;
          } 
        }
      }
    }
  }
}