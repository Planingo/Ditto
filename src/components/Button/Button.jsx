import React from 'react';
import PropTypes from 'prop-types';
import { Button as AntdButton } from 'antd';
import './button.less'

/**
 * 
 * Documentations andt du composant : https://ant.design/components/button/
 * 
 */
export default function Button({...props }){
  return (
    <div className="storybook-button">
        <AntdButton
          {...props}
        />
    </div>
  );
};

// Button.propTypes = {
//   /**
//   * TODO property
//   */
//   activeGrid: PropTypes.bool,
//   /**
//   * TODO property
//   */
//    activeList: PropTypes.bool,
//   /**
//   * TODO property
//   */
//   switch: PropTypes.bool,
//   /**
//   * TODO property
//   */
//   block: PropTypes.bool,
//   /**
//   * TODO property
//   */
//   disabled: PropTypes.bool,
//   /**
//   * TODO property
//   */
//   danger: PropTypes.bool,
//   /**
//   * TODO property
//   */
//   ghost: PropTypes.bool,
//   /**
//   * TODO property
//   */
//   onlyIcon: PropTypes.bool,
//   /**
//   * TODO property
//   */
//    type: PropTypes.oneOf(['primary', 'dashed', 'link', 'text', 'default']),
//   /**
//   * TODO property
//   */
//   href: PropTypes.string,
//   /**
//   * TODO property
//   */
//   htmlType: PropTypes.string,
//   /**
//   * TODO property
//   */
//   className: PropTypes.string,
//   /**
//   * TODO property
//   */
//   icon: PropTypes.node,
//   /**
//   * TODO property
//   */
//   loading: PropTypes.bool,
//   /**
//   * TODO property
//   */
//   size: PropTypes.oneOf(['large', 'middle', 'small']),
//   /**
//   * TODO property
//   */
//   target: PropTypes.string,
//   /**
//   * TODO property
//   */
//   onClick: PropTypes.func,
//   /**
//   * TODO property
//   */
//   rightIcon: PropTypes.bool,
// };

// Button.defaultProps = {
//   block: false,
//   disabled: false,
//   htmlType: "button",
//   loading: false,
//   size: "middle",
//   type: 'primary',
//   onlyIcon: false,
//   rightIcon: false,
//   switch: false,
//   activeGrid: true
// };
