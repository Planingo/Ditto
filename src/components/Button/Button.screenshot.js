import { ReactScreenshotTest } from 'react-screenshot-test'
import React from 'react';
import Button from './Button';
import 'antd/dist/antd.css';

describe("Screenshot", () => {
  const button = (
    <Button type="primary">Primary Button</Button>
  );

    ReactScreenshotTest
    .create("Button")
    .viewport("Desktop", {
      width: 500,
      height: 100
    })
    .shoot("Button", button)
    .run();
})
