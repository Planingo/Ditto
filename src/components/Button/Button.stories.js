import React from 'react';

import { Button } from './Button';
import {PoweroffOutlined} from '@ant-design/icons'

export default {
  title: 'Components/Button',
  component: Button,
  argTypes: {
    size: { 
      control: {
        type: 'select',
        options: [
          'large',
          'middle',
          'small'
        ]
      },
      table: {
        defaultValue: {
          summary: 'middle'
        }
      }
    },
    shape: { 
      control: {
        type: 'select',
        options: [
          'circle',
          'round',
        ]
      },
    },
    type: { 
      control: {
        type: 'select',
        options: ['primary', 'dashed', 'link', 'text', 'default'],
      },
      table: {
        defaultValue: {
          summary: 'default'
        }
      }
    },
  },
  args: {
    children: "Ananas",
    icon: <PoweroffOutlined />
  }
};

const ButtonGlobal = (args) => <Button {...args} />;

export const Defaut = ButtonGlobal.bind({});