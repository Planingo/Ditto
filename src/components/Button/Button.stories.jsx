import React from 'react';
import { Button, PoweroffOutlined } from './Button'

export default {
  component: Button,
  title: 'Atomes/Button',
  
};

const ButtonGlobal = (args) => <Button {...args} />;

export const Default = ButtonGlobal.bind({});

Default.args = {
  children: "Ananas",
  icon: <PoweroffOutlined />
}

Default.argTypes = {
  block: {
    type: "boolean"
  },
  type: {
    type: "string"
  },
    danger: {
        type: "boolean"
    },
    disabled: {
        type: "boolean"
    },
    ghost: {
        type: "boolean"
    },
    href: {
        type: "text"
    },
    htmlType: {
        type: "text"
    },
    icon: {
        type: "array"
    },
    loading: {
      control: {
        type: "number"
      }
    },
    shape: {
        type: 'select',
        options: [
          'default',
          'circle',
          'round',
        ]
    },
    size: {
        type: 'select',
        options: [
          'large',
          'middle',
          'small'
        ]
    },
    target: {
        type: "text"
    },
    onclick: {
        type: 'select',
    },
    className: {
        type: 'text'
    }
};
