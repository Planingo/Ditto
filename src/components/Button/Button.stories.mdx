import { Meta, ArgsTable } from '@storybook/addon-docs/blocks';
import {Button} from './Button';

export default {
    title: 'Button',
    component: Button,
    argTypes: {
      backgroundColor: { control: 'color' },
    },
    parameters: {
      docs: {
        source: {
          type: 'code'
        }
      }
    }
  }

<Meta
  title="Components/Button"
  argTypes={{
    block: {
      name:'block', 
      description:'block', 
      control: {
        type: 'boolean'
      },
      table: {
        defaultValue: {
          summary: 'false'
        }
      },
    },
    danger: {
      description:'danger', 
      control: {
        type: 'boolean'
      },
      table: {
        defaultValue: {
          summary: 'false'
        }
      },
    },
    disabled: {
      description:'disabled', 
      control: {
        type: 'boolean'
      },
      table: {
        defaultValue: {
          summary: 'false'
        }
      },
    },
    ghost: {
      description:'ghost', 
      control: {
        type: 'boolean'
      },
      table: {
        defaultValue: {
          summary: 'false'
        }
      },
    },  
    href: {
      description:'href', 
      control: {
        type: 'text'
      }
    },
    htmlType: {
      description:'htmlType', 
      control: {
        type: 'text'
      },
      table: {
        defaultValue: {
          summary: 'button'
        }
      },
    },
    icon: {
      description:'icon', 
      control: {
        type: 'text'
      }
    },
    loading: {
      description:'loading', 
      control: {
        type: 'boolean | {delay : number}'
      },
      table: {
        defaultValue: {
          summary: 'false'
        }
      },
    },
    shape: {
      description:'shape', 
      control: {
        type: 'select', 
        options: ['circle', 'round']
      },
      table: {
        defaultValue: {
          summary: 'round'
        }
      },
    },
    size: {
      description:'size', 
      control: {
        type: 'select', 
        options: ['large', 'middle', 'small']
      },
      table: {
        defaultValue: {
          summary: 'middle'
        }
      },
    },
    target: {
      description:'target', 
      control: {
        type: 'text'
      }
    },
    type: {
      description:'type', 
      control: {
        type: 'select', 
        options: ['primary', 'ghost', 'dashed', 'link', 'text', 'default']
      },
      table: {
        defaultValue: {
          summary: 'primary'
        }
      },
    },
    onClick: {
      description:'onClick', 
      action: 'Click on the button'
    },
    children: {
      description:'children', 
      control: {
        type: 'text'
      },
    },
  }}
  args={{
    type: 'primary',
    size: 'middle',
    children: 'Button',
  }}
/>

### Button

Use this playground to test the button component

<Canvas>
  <Story name="Standard">
    {args => {
      return <Button {...args} />;
    }}
  </Story>
</Canvas>


<ArgsTable story="Standard" />


## Variation on type

<Story name="Type">
  {args => {
    return (
      <>
        <Button {...args} type="primary">
          Primary
        </Button>
        <Button {...args} type="ghost">
          ghost
        </Button>
        <Button {...args} type="dashed">
          dashed
        </Button>
        <Button {...args} type="link">
          link
        </Button>
        <Button {...args} type="text">
          text
        </Button>
        <Button {...args} type="default">
          default
        </Button>
      </>
    );
  }}
</Story>