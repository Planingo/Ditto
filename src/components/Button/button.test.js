import '@testing-library/jest-dom/extend-expect';
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import Button from './Button';

describe('Button component', () => {
  const textContent = 'LOGIN';
  const onClick = jest.fn();

  const button = (
    <Button onClick={e => onClick()} label='test'>{textContent}</Button>
  );

  it('Snapshot button', () => {
    const { baseElement } = render(button);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render given children', () => {
    render(button);
    expect(screen.getByText(textContent)).toBeInTheDocument();
  });

  it('clicks on button', () => {
    render(button);
    const buttonClick = screen.getByRole('button');
    fireEvent.click(buttonClick);
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});