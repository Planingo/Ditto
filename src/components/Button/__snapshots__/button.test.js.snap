// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Snapshot button 1`] = `
<body>
  <div>
    <div
      class="storybook-button"
    >
      <button
        class="ant-btn ant-btn-default"
        label="test"
        type="button"
      >
        <span>
          LOGIN
        </span>
      </button>
    </div>
  </div>
</body>
`;
